sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -571,17 +571,26 @@ def split_next(self):
             if n_samples_left < n_samples_right:
                 smallest_child = left_child_node
                 largest_child = right_child_node
+                split_bin_start = 0
+                split_bin_end = node.split_info.bin_idx + 1
             else:
                 smallest_child = right_child_node
                 largest_child = left_child_node
+                split_bin_start = node.split_info.bin_idx + 1
+                split_bin_end = self.histogram_builder.n_bins
 
             # We use the brute O(n_samples) method on the child that has the
             # smallest number of samples, and the subtraction trick O(n_bins)
             # on the other one.
             # Note that both left and right child have the same allowed_features.
             tic = time()
             smallest_child.histograms = self.histogram_builder.compute_histograms_brute(
-                smallest_child.sample_indices, smallest_child.allowed_features
+                smallest_child.sample_indices,
+                smallest_child.allowed_features,
+                node.split_info.feature_idx,
+                split_bin_start,
+                split_bin_end,
```

Comment:
`bin_idx` is not used for categorical splits. I think we need to pass `node.is_categorical` here and do not reuse the parent if the split was categorical.

Reply:
Could you explain a bit further?

Reply:
I got it, it now can deal with categorical features, see https://github.com/scikit-learn/scikit-learn/pull/26189/commits/65b852be4537063cc420a801eee96367b14b5c93.